ledger/src/blockstore_processor.rs

```diff
@@ -156,13 +156,28 @@ fn aggregate_total_execution_units(execute_timings: &ExecuteTimings) -> u64 {
     execute_cost_units
 }
 
+fn sum_additional_costs(batch: &TransactionBatch, cost_model: &CostModel) -> u64 {
+    let mut costs: u64 = 0;
+    let transactions = batch.sanitized_transactions();
+
+    for transaction in transactions {
+        let transaction_cost = cost_model.calculate_cost(transaction);
+        costs = costs.saturating_add(transaction_cost.signature_cost);
+        costs = costs.saturating_add(transaction_cost.write_lock_cost);
+        costs = costs.saturating_add(transaction_cost.data_bytes_cost);
+        costs = costs.saturating_add(transaction_cost.builtins_execution_cost);
```

Comment:
is this somewhat re-implementation cost_model logic? What is cost_model adds new cost to total_cost, would be hard to enforce updating this function. Perhaps make it cost_model function then share?